### ISSUE TITLE
Supporting Opus, the format that is 2x smaller?

### ISSUE THREAD

**Issue by fzyzcjy** [ryanheise/just_audio#607]
**Is your feature request related to a problem? Please describe.**
Hi thanks for this wonderful lib!
Now I use `aac` audio formats. But I would appreciate it if I could use `opus` format. According to my experiments, opus can reduce 2x file size while preserving (and even better) audio quality.

**Describe the solution you'd like**
Suppose opus on android and ios.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by ryanheise** [comment]
Related to #20 . It's not always an easy thing to do particularly with iOS if you use the standard decoders provided by the platform. Apple has some formats it simply doesn't want to support. That doesn't mean it can't be worked around, but it may be easier to just use the formats that each platform natively supports.

**Comment by fzyzcjy** [comment]
@ryanheise Thanks. But imho `opus` is twice smaller without losing quality, so it is really desireable. For example:

* Users pay half of the money for the data usage when uploading their audio records
* Users wait for half of the time when uploading
* Users need half of time and monry for downloading (e.g. when they listen to this audio record)
* Programmers pay half of the money to store the audio records in server
* Programmers pay half of the cpu/memory to receive http file upload requests

**Comment by ryanheise** [comment]
Yes, you said that. But it's not easy to do. If you would like to contribute a pull request and do the hard work, you are welcome to help.

**Comment by fzyzcjy** [comment]
@ryanheise I will try to do so when I am free ;)
